TicTacToe.Foundation/WinningStates/WinningStateColumn.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System.Linq;
+using System.Collections.Generic;
+using TicTacToe.Foundation.Interfaces;
+
+namespace TicTacToe.Foundation.WinningStates
+{
+    public class WinningStateColumn : WinningState
+    {
+        public WinningStateColumn(IBoard board, int columnToCheck)
+        {
+            _cells = (IReadOnlyCollection<ICell>)Enumerable.Range(0, board.BoardSize)
```

Comment:
Please, use board as cells collection source.
Please, move filtering to base class, pass only predicate from child constructor.